package/cpp/api/JsiSkTextStyle.h

```diff
@@ -0,0 +1,185 @@
+      retVal.setDecorationThicknessMultiplier(propValue.asNumber());
+    }
+    if (object.hasProperty(runtime, "decorationStyle")) {
+      auto propValue = object.getProperty(runtime, "decorationStyle");
+      retVal.setDecorationStyle(
+          static_cast<para::TextDecorationStyle>(propValue.asNumber()));
+    }
+    if (object.hasProperty(runtime, "fontFamilies")) {
+      auto propValue = object.getProperty(runtime, "fontFamilies")
+                           .asObject(runtime)
+                           .asArray(runtime);
+      auto size = propValue.size(runtime);
+      std::vector<SkString> families(size);
+      for (size_t i = 0; i < size; ++i) {
+        families[i] = propValue.getValueAtIndex(runtime, i)
+                          .asString(runtime)
+                          .utf8(runtime)
+                          .c_str();
+      }
+      retVal.setFontFamilies(families);
+    }
+    if (object.hasProperty(runtime, "fontFeatures")) {
+      auto propValue = object.getProperty(runtime, "fontFeatures")
+                           .asObject(runtime)
+                           .asArray(runtime);
+      auto size = propValue.size(runtime);
+      retVal.resetFontFeatures();
+      for (size_t i = 0; i < size; ++i) {
+        auto element = propValue.getValueAtIndex(runtime, i).asObject(runtime);
+        auto name = element.getProperty(runtime, "name")
+                        .asString(runtime)
+                        .utf8(runtime);
+        auto value = element.getProperty(runtime, "value").asNumber();
+        retVal.addFontFeature(SkString(name), value);
+      }
+    }
+    if (object.hasProperty(runtime, "fontSize")) {
+      auto propValue = object.getProperty(runtime, "fontSize");
+      retVal.setFontSize(propValue.asNumber());
+    }
+    if (object.hasProperty(runtime, "fontStyle")) {
+      auto propValue =
+          object.getProperty(runtime, "fontStyle").asObject(runtime);
+      auto weight = static_cast<SkFontStyle::Weight>(
+          propValue.hasProperty(runtime, "weight")
+              ? propValue.getProperty(runtime, "weight").asNumber()
+              : SkFontStyle::Weight::kNormal_Weight);
+
+      auto width = static_cast<SkFontStyle::Width>(
+          propValue.hasProperty(runtime, "width")
+              ? propValue.getProperty(runtime, "width").asNumber()
+              : SkFontStyle::Width::kNormal_Width);
+
+      auto slant = static_cast<SkFontStyle::Slant>(
+          propValue.hasProperty(runtime, "slant")
+              ? propValue.getProperty(runtime, "slant").asNumber()
+              : SkFontStyle::Slant::kUpright_Slant);
+
+      retVal.setFontStyle(SkFontStyle(weight, width, slant));
+    }
+    if (object.hasProperty(runtime, "foregroundColor")) {
+      auto propValue = object.getProperty(runtime, "foregroundColor");
+      SkPaint p;
+      p.setColor(JsiSkColor::fromValue(runtime, propValue));
+      retVal.setForegroundColor(p);
```

Comment:
It would be good to support passing an `SkPaint` here instead of a color – since we already have the `color` prop and `foregroundColor` takes precedence, this would be a good way to customize the paint (eg. draw with stroke only, or add filters etc). Would make sense to change the prop name in that case, or add a new prop `foregroundPaint` to support this.

Reply:
Actually, just realized the skia prop itself is called `foregroundColor` – a bit confusing but it would be better to just follow the skia API and keep the prop name.

The reason for this: I was trying to figure out how to draw a stroked paragraph text, and I don't see any other way (is there?).

Reply:
Instead of adding a specific pushPaintStyle I made the pushStyle look like this:

```js
/**
   * Pushes a text-style to the builder
   * @param style Style to push
   * @param foregroundPaint Foreground paint object
   * @param backgroundPaint Background paint object
   * @returns The builder
   */
  pushStyle: (
    style: SkTextStyle,
    foregroundPaint?: SkPaint,
    backgroundPaint?: SkPaint
  ) => SkParagraphBuilder;
```

Reply:
Good catch from you, @canfan!